### ISSUE TITLE
SDL Controller subsystems initialized on a per-controller basis

### ISSUE THREAD

**Issue by DiscoStarslayer** [xqemu/xqemu#210]
Currently we initialize the SDL sub systems for [`xid-sdl.c` on a per instance basis](https://github.com/xqemu/xqemu/blob/master/hw/xbox/xid-sdl.c#L491).

Following the guide written here: https://people.redhat.com/~thuth/blog/qemu/2018/09/10/instance-init-realize.html we initialize the subsystems once using class init.

**Comment by mborgerson** [comment]
I agree that it would be better to just do subsystem init once during startup.

However, I don't think it is a major problem (that is, causing side effects) having it done in the instance realization. According to [the docs](https://wiki.libsdl.org/SDL_InitSubSystem):

> Subsystem initialization is ref-counted, you must call SDL_QuitSubSystem() for each SDL_InitSubSystem() to correctly shutdown a subsystem manually (or call SDL_Quit() to force shutdown). If a subsystem is already loaded then this call will increase the ref-count and return.

Of course, another issue with the current code is that we do not call `SDL_QuitSubSystem`. Perhaps subsystem init in the realization function can be retained and `SDL_QuitSubSystem` can be added to  `usb_xbox_gamepad_unrealize`, putting the reference counting to use. But again, we really only need to init the subsystem once at startup.
